src/repo.rs

```diff
@@ -262,72 +282,112 @@ impl<T: NetworkAdapter + 'static> Repo<T> {
+                let mut needs_flush = false;
+                let mut discard = false;
+                loop {
+                    let pending_messages = self
+                        .pending_messages
+                        .entry(repo_id.clone())
+                        .or_insert_with(Default::default);
+                    if pending_messages.is_empty() {
+                        break;
+                    }
+                    let waker = waker_ref(&self.sink_waker);
+                    let pinned_sink = Pin::new(&mut network_adapter);
+                    let result = pinned_sink.poll_ready(&mut Context::from_waker(&waker));
+                    match result {
+                        Poll::Pending => break,
+                        Poll::Ready(Ok(())) => {
+                            let pinned_sink = Pin::new(&mut network_adapter);
+                            let result = pinned_sink.start_send(
+                                pending_messages
+                                    .pop_front()
+                                    .expect("Empty pending messages."),
+                            );
+                            if result.is_err() {
+                                discard = true;
+                                needs_flush = false;
+                                break;
+                            }
+                            needs_flush = true;
+                        }
+                        Poll::Ready(Err(_)) => {
+                            discard = true;
+                            needs_flush = false;
+                            break;
+                        }
                     }
-                    needs_flush = true;
                 }
-                Poll::Ready(Err(_)) => return,
-            }
-        }
 
-        // Flusk the sink if any messages have been sent.
-        if needs_flush {
-            let waker = waker_ref(&self.sink_waker);
-            let pinned_sink = Pin::new(&mut self.network_adapter);
-            let _ = pinned_sink.poll_flush(&mut Context::from_waker(&waker));
-        }
+                // Flusk the sink if any messages have been sent.
+                if needs_flush {
+                    let waker = waker_ref(&self.sink_waker);
+                    let pinned_sink = Pin::new(&mut network_adapter);
+                    let _ = pinned_sink.poll_flush(&mut Context::from_waker(&waker));
+                }
+                discard
+            });
     }
 
     /// Handle incoming repo events(sent by repo or document handles).
     fn handle_repo_event(&mut self, event: RepoEvent) {
         match event {
             RepoEvent::NewDocHandle(repo_id, document_id, mut info) => {
-                // Send a sync message to the creator, unless it is the local repo.
+                // Send a sync message to the creator,
+                // unless it is the local repo,
+                // and all other repos we are connected with.
```

Comment:
Basically when the local client creates a new document, sync begins for all repos we are connected with.